src/main/java/thePackmaster/cards/clawpack/AbstractClawCard.java

```diff
@@ -17,10 +21,64 @@ public AbstractClawCard(String cardID, int cost, CardType type, CardRarity rarit
     }
 
     public AbstractClawCard(String cardID, int cost, CardType type, CardRarity rarity, CardTarget target) {
-        this (cardID, cost, type, rarity, target, ThePackmaster.Enums.PACKMASTER_RAINBOW);
+        this(cardID, cost, type, rarity, target, ThePackmaster.Enums.PACKMASTER_RAINBOW);
+    }
+
+
+    public static void ClawUp(int value) {
+    ClawUp(value, false);
     }
 
-    public void ClawUp(int value){
-        Wiz.applyToSelf(new ClawStrengthPower(AbstractDungeon.player, value));
+    public static void ClawUp(int value, boolean excludeClaws) {
+
+        SpireAnniversary5Mod.CLAW_SHARP_TRACKER += value;
+
+        Iterator<AbstractCard> var1 = AbstractDungeon.player.discardPile.group.iterator();
+
+        AbstractCard c;
+        while (var1.hasNext()) {
+            c = var1.next();
+            if (!(excludeClaws && (c instanceof Claw)) && c.hasTag(SpireAnniversary5Mod.CLAW)) {
+                c.baseDamage += value;
+                c.applyPowers();
+                if (c.cardsToPreview != null){
+                    if (c.cardsToPreview instanceof GhostClaw){
+                        ((GhostClaw)c).refreshDamage();
+                    }
+                }
+            }
+        }
+
+        var1 = AbstractDungeon.player.drawPile.group.iterator();
+
+        while (var1.hasNext()) {
+            c = var1.next();
+            if (!(excludeClaws && (c instanceof Claw)) && c.hasTag(SpireAnniversary5Mod.CLAW)) {
+                c.baseDamage += value;
+                c.applyPowers();
+                if (c.cardsToPreview != null){
+                    if (c.cardsToPreview instanceof GhostClaw){
+                        ((GhostClaw)c).refreshDamage();
+                    }
+                }
+            }
+
+        }
+
+        var1 = AbstractDungeon.player.hand.group.iterator();
+
+        while (var1.hasNext()) {
+            c = var1.next();
+            if (!(excludeClaws && (c instanceof Claw)) && c.hasTag(SpireAnniversary5Mod.CLAW)) {
+                c.baseDamage += value;
+                c.applyPowers();
+                if (c.cardsToPreview != null){
+                    if (c.cardsToPreview instanceof GhostClaw){
+                        ((GhostClaw)c).refreshDamage();
+                    }
+                }
+            }
+        }
```

Comment:
I can see that you took this from the decompiled GashAction (thus all the iterators). I'm sure it works just fine, but this can be written in a way that's a lot cleaner, and it's a pretty easy change to make. I think I can make a Github suggestion for code to replace this, so try the below on for size! It's much more readable.

```suggestion
         List<AbstractCard> cards = new ArrayList<>();
        cards.addAll(AbstractDungeon.player.hand.group);
        cards.addAll(AbstractDungeon.player.drawPile.group);
        cards.addAll(AbstractDungeon.player.discardPile.group);
        
        for (AbstractCard c : cards) {
            if (!(excludeClaws && (c instanceof Claw)) && c.hasTag(SpireAnniversary5Mod.CLAW)) {
                c.baseDamage += value;
                c.applyPowers();
                if (c.cardsToPreview != null){
                    if (c.cardsToPreview instanceof GhostClaw){
                        ((GhostClaw)c).refreshDamage();
                    }
                }
            }
        }
```

Reply:
I also noticed a bug in this code! For the Ghost Claw check:
```
                    if (c.cardsToPreview instanceof GhostClaw){
                        ((GhostClaw)c).refreshDamage();
                    }
```

You're checking whether `c.cardsToPreview` is a Ghost Claw, but then you're trying to cast `c`. This will crash when you use the attack potion but Claw.

I've edited my suggestion for making this code more readable to also include a fix for this.